providers/fips/self_test.c

```diff
@@ -46,7 +46,7 @@
 #define DIGEST_NAME "SHA256"
 
 static int FIPS_conditional_error_check = 1;
-static int FIPS_state = FIPS_STATE_INIT;
+static int FIPS_state = FIPS_STATE_SELFTEST;
```

Comment:
If you put this line in then please remove all the dep code..

Reply:
Where known platforms is all the platforms here that currently handle the dep

Reply:
Reworked this to fix #15322.